timemachine/cpp/src/barostat.cu

```diff
@@ -157,17 +159,17 @@ void __global__ find_group_centroids(
     const double * __restrict__ coords, // Coordinates
     const int * __restrict__ atom_idxs, // [N]
     const int * __restrict__ mol_idxs, // [N]
-    double * __restrict__ centroids // [num_molecules * 3]
+    unsigned long long * __restrict__ centroids // [num_molecules * 3]
 ) {
     const int idx = blockIdx.x*blockDim.x + threadIdx.x;
     if (idx >= N) {
         return;
     }
     const int atom_idx = atom_idxs[idx];
     const int mol_idx = mol_idxs[idx];
-    atomicAdd(centroids + mol_idx*3+0, coords[atom_idx*3+0]);
-    atomicAdd(centroids + mol_idx*3+1, coords[atom_idx*3+2]);
-    atomicAdd(centroids + mol_idx*3+2, coords[atom_idx*3+2]);
+    atomicAdd(centroids + mol_idx*3+0, FLOAT_TO_FIXED<double>(coords[atom_idx*3+0]));
```

Comment:
I just realized that this will probably be horrendously slow for the protein - since the group sizes for the centroid is huge. We can address this in a separate PR by doing reduction on this probably.